src/zarr/codecs/crc32c_.py

```diff
@@ -37,7 +38,7 @@ async def _decode_single(
         crc32_bytes = data[-4:]
         inner_bytes = data[:-4]
 
-        computed_checksum = np.uint32(crc32c(inner_bytes)).tobytes()
+        computed_checksum = np.uint32(crc32c(cast(typing_extensions.Buffer, inner_bytes))).tobytes()
```

Comment:
Isn't inner_bytes a numpy array, not a buffer here? So it needs to be converted to a buffer, not just have it's type cast? Maybe I'm misunderstanding something here...

Reply:
Correct, it is an ndarray. At least as far as my copy of mypy is concerned, that's not sufficient for this type check to pass. This might have changed in newer versions of python though.

Reply:
Are ndarrays also Buffers in reality then?

Reply:
Here's the definition of Buffer: https://github.com/python/typing_extensions/blob/d9509f902c20e3d51c62b8abe522809b4760c0ff/src/typing_extensions.py#L3316-L3342

In python 3.12, a new ABC was added, to cover the addition of a Python API to the buffer protocol: https://peps.python.org/pep-0688/

For 3.11 and earlier, ndarrays are compatible with Buffer (they implement the buffer protocol) but apparently mypy doesn't know that.